R/type-factor.R

```diff
@@ -146,17 +144,47 @@ vec_cast.factor.factor <- function(x, to, ..., x_arg = "", to_arg = "") {
     maybe_lossy_cast(out, x, to, lossy, x_arg = x_arg, to_arg = to_arg)
   }
 }
+
+#' @export
+#' @method vec_cast.factor factor
+vec_cast.factor.factor <- function(x, to, ...) {
+  fct_cast(x, to, ...)
+}
+#' @export
+#' @method vec_cast.factor ordered
+vec_cast.factor.ordered <- function(x, to, ...) {
+  fct_cast(x, to, ...)
+}
 #' @export
 #' @method vec_cast.factor character
-vec_cast.factor.character <- vec_cast.factor.factor
+vec_cast.factor.character <-function(x, to, ...) {
+  fct_cast(x, to, ...)
+}
+
 #' @export
 #' @method vec_cast.character factor
-vec_cast.character.factor <- function(x, to, ...) as.character(x)
+vec_cast.character.factor <- function(x, to, ...) {
+  as.character(x)
+}
```

Comment:
Might temporarily need `vec_cast.character.ordered()` to _technically_ be correct. But I think when we restrict `vec_cast()` that will get removed too? https://github.com/r-lib/vctrs/issues/794

Reply:
And maybe `vec_cast.ordered.character`

Reply:
oh this works because this is handled at C level right? I was puzzled for a while. I'll add "never-called" indicative methods.